Domain/Do-Svyazi.User.Domain/Chats/Chat.cs

```diff
@@ -1,19 +1,97 @@
+using Do_Svyazi.User.Domain.Exceptions;
 using Do_Svyazi.User.Domain.Roles;
 using Do_Svyazi.User.Domain.Users;
 
 namespace Do_Svyazi.User.Domain.Chats;
 
 public abstract class Chat
 {
+    private readonly List<ChatUser> _users = new ();
+    private readonly List<Role> _roles = new ();
+
     protected Chat() { }
+
+    protected Chat(string name, string description)
+    {
+        Name = name;
+        Description = description;
+    }
+
     public Guid Id { get; protected init; } = Guid.NewGuid();
-    public string Name { get; init; }
-    public string Description { get; init; }
+    public string Name { get; protected set; }
+    public string Description { get; protected set; }
 
     // public long Tag { get; init; } ??
-    public List<ChatUser> Users { get; init; } = new ();
-    public List<Role> Roles { get; init; } = new ();
+    public IReadOnlyCollection<ChatUser> Users => _users.AsReadOnly();
+    public IReadOnlyCollection<Role> Roles => _roles.AsReadOnly();
 
     protected Role BaseAdminRole { get; init; }
     protected Role BaseUserRole { get; init; }
+
+    public virtual void ChangeNameChat(string name)
+    {
+        if (string.IsNullOrWhiteSpace(name))
+            throw new ArgumentNullException("Chat name to set is null");
+
+        Name = name;
+    }
+
+    public virtual void ChangeDescriptionChat(string description)
+    {
+        if (string.IsNullOrWhiteSpace(description))
+            throw new ArgumentNullException("Chat name to set is null");
+
+        Description = description;
+    }
+
+    public List<ChatUser> GetChatUsersRole(Role role)
+    {
+        return Users.Where(user => user.Role.Equals(role)).ToList();
+    }
+
+    public ChatUser GetChatUser(string nickName)
+    {
+        return Users.SingleOrDefault(user => user.NickName == nickName)
+               ?? throw new Do_Svyazi_User_NotFoundException("User is not found in chat");
+    }
+
+    public virtual void AddChatUser(ChatUser chatUser)
+    {
+        if (chatUser is null)
+            throw new ArgumentNullException("User to set is null");
+
+        if (Users.Contains(chatUser))
+            throw new Do_Svyazi_User_InnerLogicException("This user already exists");
```

Comment:
Какой This? Ты не даёшь никакой инфы о том, что это за юзер
```suggestion
            throw new Do_Svyazi_User_InnerLogicException($"User {chatUser.nickName} already exists in chat {Name}");
```